src/shared/logs-show.c

```diff
@@ -969,16 +1001,32 @@ int output_journal(
                 OutputMode mode,
                 unsigned n_columns,
                 OutputFlags flags,
+                char **output_fields,
                 bool *ellipsized) {
 
         int ret;
+        _cleanup_set_free_ Set *fields = NULL;
         assert(mode >= 0);
         assert(mode < _OUTPUT_MODE_MAX);
 
         if (n_columns <= 0)
                 n_columns = columns();
 
-        ret = output_funcs[mode](f, j, mode, n_columns, flags);
+        if (output_fields) {
+                char **field;
+
+                fields = set_new(&string_hash_ops);
+                if (!fields)
+                        return log_oom();
+
+                STRV_FOREACH(field, output_fields) {
+                        ret = set_put(fields, *field);
+                        if (ret < 0 && ret != -EEXIST)
+                                return ret;
+                }
```

Comment:
sounds like a job for `set_put_strdupv()`...

That said, maybe we should add set_new_from_strv() or so as a generic helper function to hashmap.c that  converts an strv to a Set* the way you do it here. I figure we can make use of that elsewhere too in the long run.

Reply:
I've changed it to use `set_put_strdupv()`. Notice that now we're copying all strings, which we didn't do before.

I haven't added the new function because there's too many open questions for me about systemd's conventions (`log_oom()` in utility functions? return as first parmeter if a constructor can fail? etc). Would be much quicker to do for someone in the core team.